### ISSUE TITLE
Problem with self-referential foreign key data entry

### ISSUE THREAD

**Issue by karlcz** [informatics-isi-edu/ermrestjs#606]
I see a problem in Chaise but assume it is really in ermrestjs...

I have a table that has both `RID` and a legacy `ID` key column.  I have added a field `Source` which is a foreign key back to `RID` of the same table.  So, the table can store records where one record points to another record.

I put the foreign key name in my visible columns list for `*` context to cover `entry/edit` which is what I really care about here.  The entry field for this constraint shows up with the correct field name and tooltip, but instead of seeing a drop-down picker like for other foreign key fields, I see a rounded, dark grey box and the cursor turns into a "not allowed" circle with strikeout icon when I move it over the field.

Edit: The only thing I can see that distinguishes this failing case from other working cases is that the foreign key is a self-referential pattern involving one table, rather than pointing to a separate table of domain values.

**Comment by RFSH** [comment]
Do you have an example of this in your deployment that I can test it with? 
I tested it on `Zebrafish:Image` by adding a break point and and manipulating the annotation, it seems fine. The grey box with the "not allowed" circle suggests that the input was flagged as generated (or immutable). Is it possible that you have used the column instead of the constraint name of foreign key?

**Comment by hongsudt** [comment]
@karlcz Can you respond to @RFSH question?

**Comment by karlcz** [comment]
Sorry I forgot about this issue.

I have restored a test config on synapse-dev Image table.  There is an `edit` context visible column list and it has foreign key names for both `Image_Experimentalist_fkey` and `Image_Source Image_fkey`.  In edit mode on existing records, I see the entity-picker modal for the `Experimentalist` input field, but the grayed out problem on the `Source Image` input field.

**Comment by RFSH** [comment]
I'm not sure what the issue was when this issue was created, but now it is working properly for me. 

I don't have owner access on the catalog so I couldn't modify the annotation directly, but I was able to modify the visible-columns list by adding a breakpoint in the code. I added the `["Zebrafish", "Image_Source Image_fkey"]` constraint name to the list of visible columns and it's showing properly and it's not grayed out.

I am going to assign this to @karlcz for verification.
